src/services/transactionFilter.ts

```diff
@@ -16,29 +15,34 @@ export default class TransactionFilterService {
   constructor (
     private readonly providerService: ProviderService,
     private readonly subscriptionService: SubscriptionService,
-    private readonly broadcastService: BroadcastService
+    private readonly broadcastService: BroadcastService,
+    private readonly filterStatusService: FilterStatusService
   ) {}
 
   get provider () {
     return this.providerService.getProvider()
   }
 
   init () {
-    this.transactionFilters.push(nativeTransferToTransactionFilter)
+    this.transactionFilters.push(nativeTransferTransactionFilter)
 
     this.start()
   }
 
   async start () {
-    const latestBlock = await this.provider.getBlock('latest')
-    const blockTracker = await this.getBlockTracker()
+    while (true) {
+      const { number: toBlockNumber } = await this.provider.getBlock('latest')
 
-    await this.processBlocks(
-      blockTracker.block + 1 || latestBlock.number,
-      latestBlock.number
-    )
+      const filterStatus = await this.filterStatusService.getFilterStatus('transaction')
```

Comment:
What about the case there's no new blocks?
Is the last block is already processed, no need to continue processing. And we can do a sleep for some interval (like 1  sec)